### PR TITLE
fix(a380x/fms): use estimated landing weight for characteristic speeds

### PR DESCRIPTION
**Pull request by BlueberryKing**
Fixes #[issue_no]

## Summary of Changes

Fixes a bug where the characteristic approach speeds (F, S, GD) were calculated based on the current weight and the current altitude rather than the estimated landing weight and elevation. This caused a bug where the predictions would crash because the green dot speed was higher than 250 kts. 

## Additional context

Discord username (if different from GitHub):

## Testing instructions

1. Plan a long flight with high weights. Weights should be chosen so that green dot speed on FMS/PERF page is higher than 250 knots when on CRZ FL
2. Observe GD speed on PERF APPR page staying more or less constant throughout the flight.
3. VNAV predictions shouldn't suddenly go away if GD is over 250kts (heavy weight, at high CRZ FL)

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

**Review by 2hwk — APPROVED** [review]

**Comment by becas22** [comment]
What happens when you activate approach phase?

**Comment by tracernz** [comment]
If it's like the A320 it should use live data in approach phase.

File: fbw-a380x/src/systems/instruments/src/MFD/FMC/FmcAircraftInterface.ts [hunk 1]
```diff
@@ -1060,35 +1060,48 @@ export class FmcAircraftInterface {
...
+      alt = (alts[0].value + alts[1].value) / 2;
     } else if (adrOp.filter(Boolean).length === 1) {
       const aoas = [aoa1, aoa2, aoa3].filter((v, i) => adrOp[i]);
       aoa = aoas[0].value;
 
       const cass = [cas1, cas2, cas3].filter((v, i) => casOp[i]);
       cas = cass[0].value;
+
+      const alts = [alt1, alt2, alt3].filter((v, i) => altOp[i]);
+      alt = alts[0].value;
```
Comment (tracernz) [inline fbw-a380x/src/systems/instruments/src/MFD/FMC/FmcAircraftInterface.ts:1104]: Why not just use the FMGC Navigation class for this? It's job is to source navigation data from the relevant systems, select the appropriate sources, and eventually also run the filters to combine sources where applicable.

**Comment by BlueberryKing** [comment]
> What happens when you activate approach phase?

I will double check this.

**Review by flogross89 — APPROVED** [review]
Looks good from my side, pending the remarks from tracer.
Thank you!

**Comment by 2hwk** [comment]
Quality Assurance Report

Discord Username  : 2cas
Object of testing : #9063
Aircraft          : A380X
Tier of Testing   : 1
Date              : 17/10/2024

Testing Process: 

- [x] Plan a long flight with high weights
- [x] Observe GD speed on PERF APPR page staying more or less constant throughout the flight.
- [x] Tried to recreate same OMDB-EGLL route that caused the bug previously, was not able to reproduce.


Negatives:
 - N/A

Testing Results: 
Passed

File: fbw-a380x/src/systems/instruments/src/MFD/FMC/FmcAircraftInterface.ts (hunk 1 above)
Comment (flogross89) [reply fbw-a380x/src/systems/instruments/src/MFD/FMC/FmcAircraftInterface.ts:1104]: With the next commit, CAS alt alt are going to be taken from NavigationProvider. AoA is going to be removed, not used anymore in the FMS calculations

**Comment by flogross89** [comment]
> If it's like the A320 it should use live data in approach phase.

image
Already uses live data in approach or GA phase

**Comment by flogross89** [comment]
@tracernz I hope your remarks are addressed now

**Review by tracernz — APPROVED** [review]
💯

**Comment by danbw9** [comment]
QA Report

Discord Username : danbw9
Object of testing : https://github.com/flybywiresim/aircraft/pull/9063
Aircraft : A380X
Tier of Testing : 1
Date : 18/10/2024

Testing Process:

Planned two long flights with weights and after take off at CRZ with heavy weight close to MTOW (510T) and completed a full flight from RJBB to OMDB - VNAV crash previously observed with TOW of 470T.

Planned OMDB-YSSY and EGLL-WSSS. Flew up at CRZ APPR PERF with Green dot indicated as above 250. No VNAV crash.

Completed long haul 10hr30 mins from RJBB to OMDB. 470 T TOW gave green dot speed at 254. No VNAV crash or change in aircraft behaviour with ND routing or predictions showing 'N N N N' as was observed before. Flight could not be continued with manual HDG selections.

Observed that Green Dot speed on PERF APPR page stayed more or less constant throughout the flight.

Negatives:

N/A
Testing Results:
Passed

**Pull request merged**